add propeller docs (#34)
equinox-project/equinox-js: docs/docs/reactions/projections.md

  if (statements.length === 0) return
  const client = await pool.connect()
  try {

>     // CAUTION: do not lower the isolation level unless you are Kimi Raikkonen

Reviewer: I guess if you know you know but https://www.google.com/search?q=raikkonen+quote gives a good result at this moment in time for me!

Author: ![image](https://github.com/equinox-project/equinox-js/assets/3721521/2db9522c-fa3a-4afb-b862-44c01e47b119)

>     // CAUTION: do not lower the isolation level unless you know what you're doing

Reviewer: ```suggestion
    // CAUTION: do not lower the isolation level unless you are Kimi Raikkonen
```

Author: Yes!

    await client.query("BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE")
    for (const statement of statements) {
      await client.query(statement)
    }
    await client.query("COMMIT")
  } catch (err) {
    await client.query("ROLLBACK")
  } finally {
    client.release()
  }
}

const source = MessageDbSource.create({
  pool,
  batchSize: 500,
  categories: [Payer.CATEGORY],
  groupName: "PayerListModel",
  checkpoints,
  handler,
  tailSleepIntervalMs: 5000,
  maxConcurrentStreams: 10,

>   maxConcurrentBatches: 10,

Reviewer: maybe start using a diff value for this - 2 or 5 might be good candidates
a) >1 so nobody is waiting for a read
b) not too high so no memory wasted
c) different to the 10 for streams

ultimately if you want to have a lot of stuff buffered, the batch size and the maxReadAhead multipled together are the real number (multiplied by the number of tranches that this node is handling)

if you start heading to 10, you probably just want to up the batch size instead. Downsides:
- bigger batch sizes = less frequent checkpointing -> more rework if killed; less insight into gap/lag from an external observer -> more chunky read traffic -> potentially worse min sunny day reaction latency
- smaller batch sizes -> more memory usage, more store roundtrips, better sunny day reaction latency

The key thing allowing more read ahead can enable is to allow processing to still be reacting to new stuff despite something big/slow/randomly failing in the oldest batch holding up processing in general

This is probably worth calling out a a higher level; it all depends on whether the work being performed by the handler is heterogenous in terms of:
- mix of IO vs compute bound
- mix of high/low traffic streams
- ability of a handler to handle sequences of events efficiently relative to processing them individually

Author: Great point, I'll add a section on this and change the default in the docs to 3

Author: See `configuring.md`
